che-editors.yaml

```diff
@@ -1,373 +1,442 @@
-        attributes:
-          protocol: http
-          discoverable: false
-      - name: "theia-redirect-3"
-        public: true
-        targetPort: 13133
-        attributes:
-          protocol: http
-          discoverable: false
-      - name: terminal
-        public: true
-        targetPort: 3333
-        attributes:
-          protocol: ws
-          type: collocated-terminal
-          discoverable: false
-          secure: true
-          cookiesAuthEnabled: true
-    containers:
+  - schemaVersion: 2.1.0
+    metadata:
+      name: eclipse/che-theia/next
+      displayName: theia-ide
+      description: Eclipse Theia, get the latest release each day.
+      icon: /v3/images/eclipse-che-logo.png
+      attributes:      
+        publisher: eclipse
+        version: next
+        type: Che Editor
+        title: Eclipse Theia development version.
+        category: Editor
+        repository: 'https://github.com/eclipse/che-theia'
+        firstPublicationDate: '2019-03-07'
+        latestUpdateDate: '2021-03-18'
+    commands:
+      - id: init-container-command
+        apply:
+          component: remote-runtime-injector
+    events:
+      preStart:
+        - init-container-command
+    components:
       - name: theia-ide
-        image: "quay.io/eclipse/che-theia:next"
-        env:
-          - name: THEIA_PLUGINS
-            value: local-dir:///plugins
-          - name: HOSTED_PLUGIN_HOSTNAME
-            value: 0.0.0.0
-          - name: HOSTED_PLUGIN_PORT
-            value: "3130"
-          - name: THEIA_HOST
-            value: 127.0.0.1
-        volumeMounts:
-          - name: plugins
-            path: "/plugins"
-          - name: theia-local
-            path: "/home/theia/.theia"
-        mountSources: true
-        ports:
-          - exposedPort: 3100
-          - exposedPort: 3130
-          - exposedPort: 13131
-          - exposedPort: 13132
-          - exposedPort: 13133
-        memoryLimit: "512M"
-        cpuLimit: 1500m
-        cpuRequest: 100m
+        container:
+          image: 'quay.io/eclipse/che-theia:next'
+          env:
+            - name: THEIA_PLUGINS
+              value: 'local-dir:///plugins'
+            - name: HOSTED_PLUGIN_HOSTNAME
+              value: 0.0.0.0
+            - name: HOSTED_PLUGIN_PORT
+              value: '3130'
+            - name: THEIA_HOST
+              value: 0.0.0.0
+          volumeMounts:
+            - name: plugins
+              path: /plugins
+            - name: theia-local
+              path: /home/theia/.theia
+          mountSources: true
+          memoryLimit: 512M
+          cpuLimit: 1500m
+          cpuRequest: 100m
+          endpoints:
+            - name: theia
+              attributes:
+                type: ide
+                cookiesAuthEnabled: true
+                discoverable: false
+              targetPort: 3100
+              exposure: public
+              secure: false
+              protocol: http
```

Comment:
it's secured:true protocol http for che-server

and secure:false with https protocol for DevWorkspace

so as long as meta.yaml and devfile.yaml are generated correctly I'm fine

Reply:
Which one should I keep, then?

Reply:
maybe keep what is used by devworkspace now ( so secured false and protocol https) and when converting to meta.yaml, use secured true if protocol is ending with a s